node/request.js

```diff
@@ -195,20 +204,28 @@ TChannelRequest.prototype.shouldRetry = function shouldRetry(err, res, arg2, arg
         return false;
     }
 
+    if (self.options.retryFlags.never) {
+        return false;
+    }
+
     if (err) {
         switch (err.type) {
             case 'tchannel.bad-request':
             case 'tchannel.canceled':
                 return false;
 
-            case 'tchannel.socket':
-            case 'tchannel.timeout':
             case 'tchannel.busy':
             case 'tchannel.declined':
-            case 'tchannel.unexpected':
-            case 'tchannel.protocol':
                 return true;
```

Comment:
cc @mranney @mmihic does this seem like a missed flag? or intentionally left out?